doc/reference/std/errors.md

```diff
@@ -316,12 +316,22 @@ If the exception object is an instance of `StackTrace`, then unless
 there is already a continuation because this is a re-raise, the
 continuation is captured and stored in the `continuation` slot.
 
-### raise-bad-argument
+### check-argument
 ```scheme
-(raise-bad-argument where expectation bad)
+(check-argument expr expectation argument)
+  expr := a boolean expression
+  expectation := a string describing the expectation
+  argument := the argument being checked
 ```
 
-Raises a `BadArgument` condition.
+Evaluations the boolean expression `expr` and raise a contract violation if it is `#f`.
```

Comment:
What if the contract violation deal with two arguments? Can we have a list of irritants instead of just one?
`(check-argument (> a b) "a greater than b" [a b])`

Reply:
expr is an expression, it can take as many arguments as it likes.

Reply:
and yes, we can pass extra irritants.